File: pythonProject/Usuario.py
from datetime import date

class Usuario:
    def __init__(self, id=None, nombre=None, apellido=None, correo=None, contrasena=None, fecha_nacimiento=None, libros_descargados=None):
        self.id = id
        self.nombre = nombre
        self.apellido = apellido
        self.correo = correo
        self.contrasena = contrasena
        self.fecha_nacimiento = fecha_nacimiento
        self.libros_descargados = libros_descargados

    def __str__(self):
        autores_str = ', '.join(f"{autor.nombre} {autor.apellido}" for autor in self.autores)
        return f"Libro{{id={self.id}, titulo='{self.titulo}', autores=[{autores_str}], genero='{self.genero}'}}"

File: javascriptProject/Usuario.js
class Usuario {
    constructor(id = null, nombre = null, apellido = null, correo = null, contrasena = null, fechaNacimiento = null, librosDescargados = []) {
        this.id = id;
        this.nombre = nombre;
        this.apellido = apellido;
        this.correo = correo;
        this.contrasena = contrasena;
        this.fechaNacimiento = fechaNacimiento;
        this.librosDescargados = librosDescargados;
    }

    getId() {
        return this.id;
    }

    setId(id) {
        this.id = id;
    }

    getNombre() {
        return this.nombre;
    }

    setNombre(nombre) {
        this.nombre = nombre;
    }

    getApellido() {
        return this.apellido;
    }

    setApellido(apellido) {
        this.apellido = apellido;
    }

    getCorreo() {
        return this.correo;
    }

    setCorreo(correo) {
        this.correo = correo;
    }

    getContrasena() {
        return this.contrasena;
    }

    setContrasena(contrasena) {
        this.contrasena = contrasena;
    }

    getFechaNacimiento() {
        return this.fechaNacimiento;
    }

    setFechaNacimiento(fechaNacimiento) {
        this.fechaNacimiento = fechaNacimiento;
    }

    getLibrosDescargados() {
        return this.librosDescargados;
    }

    setLibrosDescargados(librosDescargados) {
        this.librosDescargados = librosDescargados;
    }

    toString() {
        return `Usuario{id=${this.id}, nombre='${this.nombre}', apellido='${this.apellido}', correo='${this.correo}', contrasena='${this.contrasena}', fechaNacimiento=${this.fechaNacimiento}, librosDescargados=[${this.librosDescargados.join(", ")}]}`;
    }
}

module.exports = Usuario;